pkg/app/sensor/app.go

```diff
@@ -381,6 +388,36 @@ func startMonitor(
 	return res, nil
 }
 
+func runEventReporter(
+	ctx context.Context,
+	eventChan <-chan event.Message,
+	errorChan <-chan error,
+) {
+	errutil.FailOn(fsutil.Touch(defaultEventsFileName))
+
+	out, err := os.OpenFile(defaultEventsFileName, os.O_APPEND|os.O_WRONLY, 0644)
+	errutil.FailOn(err)
+	defer out.Close()
+
+	log.Debug("sensor: event collector - waiting for events...")
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("sensor: event collector - done...")
+			return
+		case evt := <-eventChan:
+			log.WithField("event", evt).Info("sensor: event collector - dumping event")
+			writeJSON(evt, out)
+		case err := <-errorChan:
+			log.WithError(err).Warn("sensor: event collector - dumping error")
+			writeJSON(struct {
+				Error string `json:"error"`
```

Comment:
we need to use the event structure we have for the other events: `event.Message{Name: event.Error, Data: err.Error()}`

Reply:
Done!